pkg/scheduler/scheduler.go

```diff
@@ -199,7 +199,12 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 	entries := s.nominate(ctx, headWorkloads, snapshot)
 
 	// 4. Create iterator which returns ordered entries.
-	var iterator entryIterator = makeIterator(entries, s.workloadOrdering, s.fairSharing.Enable)
+	var iterator entryIterator
```

Comment:
nit, I would fold this dispatching into the preexisting `makeIterator` to offload the main scheduling function. 
The function is complex already, so minimizing "technical" distractions is preferred imo.